lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -874,20 +866,22 @@ SliceEncodingAttr::toLinearLayout(ArrayRef<int64_t> shape) const {
   return ret;
 }
 
-std::optional<LinearLayout>
+LinearLayout
 TritonGPUDialect::toLinearLayout(ArrayRef<int64_t> shape, Attribute layout,
                                  std::optional<int32_t> elemBitWidth) {
   CacheKey key{std::vector<int64_t>(shape.begin(), shape.end()), layout,
                elemBitWidth};
-  auto result = llCache.get(key);
-  if (result.has_value()) {
-    return result;
+  if (auto result = llCache.get(key)) {
+    return *result;
   }
 
   // Layouts are distributed or shared in triton core
+  // To add a new layout add an else-if clause
+  LinearLayout result = LinearLayout::empty();
```

Comment:
Does LinearLayout not have a default constructor?

Reply:
It doesn't

Reply:
Okay, but I'm really asking _why_ it doesn't have a default constructor. That's really unidiomatic, so I presume there must be a good reason.

Reply:
I don't know why doesn't it have a default constructor. IMO it should have one. cc @jlebar for historical context

Reply:
I suspect that the usefulness of an empty LL was not immediately obvious to me, so at first there was no default constructor *or* `::empty()` call.  But 🤷 it wasn't an important design point that I remember.